src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -341,13 +341,20 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
         // assumes all same case.        
         private bool ArgumentMatch(string arg, string formal)
         {
-            if (arg[0] != '/' && arg[0] != '-')
+            if (arg[0] != '-' || arg.Length==1)
             {
                 return false;
             }
-
-            arg = arg.Substring(1);
-            return (arg == formal || (arg.Length == 1 && arg[0] == formal[0]));
+            if (arg[1] == '-')
+            {
+                arg = arg.Substring(2);
+                return arg == formal;
+            }
+            else
+            {
+                arg = arg.Substring(1);
+                return arg.Length == 1 && arg[0] == formal[0];
```

Comment:
What is to check here? Do we need check if arg== formal. I feel we only need one return after if/else and have a line similar as https://github.com/dotnet/corefx/blob/master/src/Microsoft.XmlSerializer.Generator/src/Sgen.cs#L350